File: src/Frontend/projects/core/src/lib/models/dto/calculator/deal.ts
import { DiscountParams } from './discount-params';
import { HeadDiscountRequest } from './head-discount-request';

export class Deal {
    public id: number;
    public contractor: string;
    public managerId: number;
    public managerName: string;
    public createDate: Date;
    public completeDate: Date;
    public headDiscount: number;
    public ceoDiscount: number;
    public brand: string;
    public volume: number;
    public partnerNameOnMarket: string;
    public currency: string;
    public installmentLimits: number[] = [];
    public discount: number;
    public netCost: number;
    public status: DealStatus;
    public discountParams: DiscountParams;
    public headDiscountRequests: HeadDiscountRequest[] = [];
    public orderFormId: number;
    public contractId: number;
    public comment: string;
    public dealMarginality: number;
    public managerMarginality: number;
    public upload1cTime: Date;
    public seasonId: number;
    public totalProductsCount: number;
    public delivery: DeliveryType;
    public productType: ProductType;

    static fromJSON(obj: any) {
        if (!obj) {
            return null;
        }

        if (!obj.delivery) {
            obj.delivery = null;
        }

        return Object.assign(
            new Deal(),
            obj,
            {
                createDate: obj.createDate ? new Date(obj.createDate) : null,
                completeDate: obj.completeDate ? new Date(obj.completeDate) : null,
                upload1cTime: obj.upload1cTime ? new Date(obj.upload1cTime) : null,
                discountParams: obj.discountParams ? DiscountParams.fromJSON(obj.discountParams) : null,
                headDiscountRequests: obj.headDiscountRequests ? obj.headDiscountRequests.map((r) => HeadDiscountRequest.fromJSON(r)) : [],
            },
        );
    }
}

export enum DealStatus {
    NotConfirmed = 1,
    Confirmed = 2,
    OnPayment = 3,
}

export enum DeliveryType {
    SelfDelivery = 1,
    OurTransportServiceToClient = 2,
    OurTransportServiceToCarrier = 3,
    CarrierFromOurStock = 4,
    AsTransportServiceWants = 5,
}

export enum ProductType {
    Product = 1,
    Sample = 2,
    SampleNotForSale = 3,
}

File: src/Backend/Common/Models/Deals/Deal.cs
using System;
using System.Collections.Generic;
using Company.Common.Enums;

namespace Company.Common.Models.Deals
{
    public class Deal
    {
        public int Id { get; set; }

        public int OrderFormId { get; set; }

        public int ContractId { get; set; }

        public int SeasonId { get; set; }

        public int BrandId { get; set; }

        public int ManagerId { get; set; }

        public int DeleterId { get; set; }

        public int BrandMix { get; set; }

        public string Contractor { get; set; }

        public string Tin { get; set; }

        public string Rrc { get; set; }

        public string Brand { get; set; }

        public string PartnerNameOnMarket { get; set; }

        public string Comment { get; set; }

        public DeliveryType? Delivery { get; set; }

        public ProductType ProductType { get; set; }

        public DateTime CreateDate { get; set; }

        public DateTime? DeleteTime { get; set; }

        public DateTime? CompleteDate { get; set; }

        public DateTime? Upload1cTime { get; set; }

        public float Volume { get; set; }

        public float VolumePure { get; set; }

        public float NetCost { get; set; }

        public float NetCostPure { get; set; }

        public float Discount { get; set; }

        public float DealMarginality { get; set; }

        public float ManagerMarginality { get; set; }

        public DealStatus DealStatus { get; set; }

        public PartnersType PartnersType { get; set; }

        public bool Repeatedness { get; set; }

        public ICollection<DealProduct> DealProducts { get; set; } = new List<DealProduct>();

        public ICollection<DiscountParams> DiscountParams { get; set; } = new List<DiscountParams>();

        public ICollection<HeadDiscountRequest> HeadDiscountRequests { get; set; } = new List<HeadDiscountRequest>();

        public ICollection<DealFile> DealFiles { get; set; } = new List<DealFile>();
    }
}